Riot/Modules/Room/RoomViewController.m

```diff
@@ -1442,11 +1486,28 @@ - (void)refreshRoomInputToolbar
     }
 }
 
-- (void)setInputToolBarSendMode:(RoomInputToolbarViewSendMode)sendMode
+- (void)setInputToolBarSendMode:(RoomInputToolbarViewSendMode)sendMode forEventWithId:(NSString *)eventId
 {
     if (self.inputToolbarView && [self.inputToolbarView isKindOfClass:[RoomInputToolbarView class]])
     {
         RoomInputToolbarView *roomInputToolbarView = (RoomInputToolbarView*)self.inputToolbarView;
+        if (eventId)
+        {
+            MXEvent *event = [self.roomDataSource eventWithEventId:eventId];
+            MXRoomMember * roomMember = [self.roomDataSource.roomState.members memberWithUserId:event.sender];
+            if (roomMember)
```

Comment:
```suggestion
            if (roomMember.displayname.length)
```

because `displayname` is nullable.

You could use
`[self.roomDataSource.roomState.members memberName.sender]`, which is the the method used in the timeline.